src/main/java/pl/javastart/task/FileUtils.java

```diff
@@ -0,0 +1,52 @@
+package pl.javastart.task;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+
+public class FileUtils {
+
+    static BigDecimal sumProductsInEuro(List<Product> products, List<Currency> currencies) {
+        BigDecimal sum = BigDecimal.valueOf(0);
+        for (Product product : products) {
+            for (Currency currency : currencies) {
+                if (product.getCurrency().equals(currency.getCurrency())) {
+                    BigDecimal valueInEuro = product.getPrice().divide(currency.getValue(), RoundingMode.UP);
+                    sum = sum.add(valueInEuro);
+                }
+            }
```

Comment:
W całej tej klasie powtarza Ci się schemat wyszukiwania waluty dla określonego produktu. Wydziel to do osobnej metody